pkg/apis/pipeline/v1beta1/result_types.go

```diff
@@ -38,6 +38,22 @@ type TaskResult struct {
 	Value *ResultValue `json:"value,omitempty"`
 }
 
+// StepResult used to describe the results of a step action
+type StepResult struct {
```

Comment:
Why use v1 for v1alpha1 but define a separate type for v1beta1?

I don't have a strong preference for one over the other if all of this is going to be considered alpha, but let's stay consistent.

Reply:
Referencing the v1 type is fine - I'm more curious why for v1alpha1 we're referencing v1, but for v1beta1 we're using a v1beta1 type. 🤔 

e.g. why not have v1beta1 use v1, or v1alpha1 use the less-stable v1beta1, or its own v1alpha1 like it was before? 🤔

Reply:
Oh I see. The reason it is in v1beta1 and v1 is so that we can use StepActions with both v1 and v1beta1 Task CRDs. There are conversion funcs between the two. So we have, `v1beta1.StepResult` and `v1.StepResult` so that both API versions can reference `StepActions` and the corresponding features.

We could map `StepAction's` `Results` to `v1beta1.StepResults` or `v1.StepResults`. I just chose `v1.StepResults` because that's less change in the long run and also consistent with what we did for StepAction's Params.

Reply:
Fixed. Even v1beta1.Step.Results points to `v1.StepResults`. So there are no multiple.

Thanks for the suggestion!